src/bindings/python/src/pyopenvino/graph/shape.cpp

```diff
@@ -40,10 +40,24 @@ void regclass_graph_Shape(py::module m) {
     shape.def("__setitem__", [](ov::Shape& self, size_t key, ov::Dimension d) {
         self[key] = d.get_length();
     });
-    shape.def("__getitem__", [](const ov::Shape& v, size_t key) {
+    shape.def("__getitem__", [](const ov::Shape& v, int key) {
+        if (key < 0)
+            key += v.size();
         return v[key];
     });
 
+    shape.def("__getitem__", [](const ov::Shape& v, py::slice slice) {
```

Comment:
You're right, it's not necessary. I was passing it by value because I wasn't sure if I would modify the object, but since I don't modify it in the function, it's safe to pass it by reference. Thank you very much.